### ISSUE TITLE
ckan crash on debian unstable

### ISSUE THREAD

**Issue by talpadk** [KSP-CKAN/CKAN#2664]
Background
----------

**CKAN Version:** v1.25.4+7413c232f2ff

**KSP Version:** 1.6.1

**Mono Version:** Mono JIT compiler version 5.18.0.240 (Debian 5.18.0.240+dfsg-1 Fri Jan 18 17:30:45 UTC 2019)

**Mono-complete installed:** Yes, mono-complete 5.18.0.240+dfsg-1

**Operating System:** Debian (Unstable, just upgraded)

**Have you made any manual changes to your GameData folder (i.e., not via CKAN)?** No, I have just unziped KSP and copied ckan.exe into it

Problem
-------

**What steps did you take in CKAN?** ran "mono ckan.exe" inside the ksp folder

**What did you expect to happen?** ckan starting

**What happened instead?** ckan crashed 
```

mono ckan.exe 

Unhandled Exception:
System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x0001c] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0002a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at System.IO.CoreFXFileSystemWatcherProxy.<StartDispatching>m__0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001d3] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
   --- End of inner exception stack trace ---
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001f8] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00007] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.Start () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x00049] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at CKAN.NetFileCache..ctor (System.String path) [0x000b6] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.NetFileCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00000] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.NetModuleCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00006] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager.TrySetupCache (System.String path, System.String& failureReason) [0x0003d] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager.LoadInstancesFromRegistry () [0x000ad] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager..ctor (CKAN.IUser user, CKAN.IWin32Registry win32_registry) [0x00028] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.KSPManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x0015b] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00091] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x0001c] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0002a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at System.IO.CoreFXFileSystemWatcherProxy.<StartDispatching>m__0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001d3] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
   --- End of inner exception stack trace ---
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001f8] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00007] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.Start () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x00049] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at CKAN.NetFileCache..ctor (System.String path) [0x000b6] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.NetFileCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00000] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.NetModuleCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00006] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager.TrySetupCache (System.String path, System.String& failureReason) [0x0003d] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager.LoadInstancesFromRegistry () [0x000ad] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager..ctor (CKAN.IUser user, CKAN.IWin32Registry win32_registry) [0x00028] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.KSPManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x0015b] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00091] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
```

**Comment by HebaruSan** [comment]
CKAN can't cause `System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents` to have no body. :)

Something's wrong with your Mono install. Try uninstalling/reinstalling, make sure it's the right version, make sure there were no errors installing, try other Mono apps, etc.

**Comment by dkm** [comment]
I can confirm the problem. I have tried to install mono from the https://www.mono-project.com/download/stable/#download-lin-debian page, but packages are already up-to-date in debian archive.

I have also installed the full mono metapackage that should install everything possible : `mono-complete`.

```
Unhandled Exception:
System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x0001c] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0002a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at System.IO.CoreFXFileSystemWatcherProxy.<StartDispatching>m__0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001d3] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
   --- End of inner exception stack trace ---
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001f8] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00007] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.FileSystemWatcher.Start () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x00049] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at CKAN.NetFileCache..ctor (System.String path) [0x000b6] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.NetFileCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00000] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.NetModuleCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00006] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.KSPManager.TrySetupCache (System.String path, System.String& failureReason) [0x0003d] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.KSPManager.LoadInstancesFromRegistry () [0x000ad] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.KSPManager..ctor (CKAN.IUser user, CKAN.IWin32Registry win32_registry) [0x00028] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.CmdLine.MainClass.Execute (CKAN.KSPManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x0015b] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00091] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x0001c] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0002a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at System.IO.CoreFXFileSystemWatcherProxy.<StartDispatching>m__0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001d3] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
   --- End of inner exception stack trace ---
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001f8] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00007] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.FileSystemWatcher.Start () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x00049] in <a9a08e39ba304bd0a84c49bd158dfc02>:0
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at CKAN.NetFileCache..ctor (System.String path) [0x000b6] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.NetFileCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00000] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.NetModuleCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00006] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.KSPManager.TrySetupCache (System.String path, System.String& failureReason) [0x0003d] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.KSPManager.LoadInstancesFromRegistry () [0x000ad] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.KSPManager..ctor (CKAN.IUser user, CKAN.IWin32Registry win32_registry) [0x00028] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.CmdLine.MainClass.Execute (CKAN.KSPManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x0015b] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00091] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
```

```
 mono --version
Mono JIT compiler version 5.18.0.240 (Debian 5.18.0.240+dfsg-1 Fri Jan 18 17:30:45 UTC 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug
        Interpreter:   yes
        LLVM:          supported, not enabled.
        Suspend:       preemptive
        GC:            sgen (concurrent by default)
```

**Comment by nsprangers** [comment]
Same issue. Reinstalled mono-complete from mono-project.com repo. Perhaps a recent mono update introduced a bug?

```
$ ckan

Unhandled Exception:
System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x0001c] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0002a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at System.IO.CoreFXFileSystemWatcherProxy.<StartDispatching>m__0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001d3] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
   --- End of inner exception stack trace ---
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001f8] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00007] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.Start () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x00049] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at CKAN.NetFileCache..ctor (System.String path) [0x000b6] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.NetFileCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00000] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.NetModuleCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00006] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager.TrySetupCache (System.String path, System.String& failureReason) [0x0003d] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager.LoadInstancesFromRegistry () [0x000ad] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager..ctor (CKAN.IUser user, CKAN.IWin32Registry win32_registry) [0x00028] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.KSPManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x0015b] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00091] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
[ERROR] FATAL UNHANDLED EXCEPTION: System.InvalidOperationException: object_op ---> System.BadImageFormatException: Method has no body
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEvents () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.StartRaisingEventsIfNotDisposed () [0x0001c] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x0002a] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.CoreFX.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at System.IO.CoreFXFileSystemWatcherProxy.<StartDispatching>m__0 (System.IO.CoreFX.FileSystemWatcher internal_fsw, System.IO.FileSystemWatcher fsw) [0x00048] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001d3] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
   --- End of inner exception stack trace ---
  at System.IO.CoreFXFileSystemWatcherProxy.Operation (System.Action`4[T1,T2,T3,T4] map_op, System.Action`2[T1,T2] object_op, System.Object handle, System.Action`2[T1,T2] cancel_op) [0x001f8] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.CoreFXFileSystemWatcherProxy.StartDispatching (System.Object handle) [0x00007] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.Start () [0x00024] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at System.IO.FileSystemWatcher.set_EnableRaisingEvents (System.Boolean value) [0x00049] in <a9a08e39ba304bd0a84c49bd158dfc02>:0 
  at (wrapper remoting-invoke-with-check) System.IO.FileSystemWatcher.set_EnableRaisingEvents(bool)
  at CKAN.NetFileCache..ctor (System.String path) [0x000b6] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.NetFileCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00000] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.NetModuleCache..ctor (CKAN.KSPManager mgr, System.String path) [0x00006] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager.TrySetupCache (System.String path, System.String& failureReason) [0x0003d] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager.LoadInstancesFromRegistry () [0x000ad] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.KSPManager..ctor (CKAN.IUser user, CKAN.IWin32Registry win32_registry) [0x00028] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.CmdLine.MainClass.Execute (CKAN.KSPManager manager, CKAN.CmdLine.CommonOptions opts, System.String[] args) [0x0015b] in <2c2371fb3f2949b2a6dfae930758e6ee>:0 
  at CKAN.CmdLine.MainClass.Main (System.String[] args) [0x00091] in <2c2371fb3f2949b2a6dfae930758e6ee>:0
```
```
$ mono --version
Mono JIT compiler version 5.18.0.240 (Debian 5.18.0.240+dfsg-1 Fri Jan 18 17:30:45 UTC 2019)
Copyright (C) 2002-2014 Novell, Inc, Xamarin Inc and Contributors. www.mono-project.com
        TLS:           __thread
        SIGSEGV:       altstack
        Notifications: epoll
        Architecture:  amd64
        Disabled:      none
        Misc:          softdebug 
        Interpreter:   yes
        LLVM:          supported, not enabled.
        Suspend:       preemptive
        GC:            sgen (concurrent by default)
```
```
$ uname -a
Linux ysera 4.19.0-2-amd64 #1 SMP Debian 4.19.16-1 (2019-01-17) x86_64 GNU/Linux
```

**Comment by Cyclic3** [comment]
I tried using a docker image to build the program from scratch (basically switching the image to use debian:sid, and added `mono-complete` to the apt-get install list), and this was the result:
```
Removing intermediate container 1c749bb62f14
 ---> 7d55014cbaf1
Step 8/17 : RUN mkdir /kspdir
 ---> Running in 21350046e52c
Removing intermediate container 21350046e52c
 ---> 2c3d517aac7c
Step 9/17 : VOLUME ["/kspdir"]
 ---> Running in 4e5b23ba9ce8
Removing intermediate container 4e5b23ba9ce8
 ---> 81ef7c74719c
Step 10/17 : COPY . /source
 ---> a2c72232117f
Step 11/17 : WORKDIR /source
 ---> Running in 007579f3eda6
Removing intermediate container 007579f3eda6
 ---> 29e042369c8b
Step 12/17 : ARG config
 ---> Running in fc02047d999e
Removing intermediate container fc02047d999e
 ---> 59da5e7cd367
Step 13/17 : ENV config ${config:-Release}
 ---> Running in 691d0d008289
Removing intermediate container 691d0d008289
 ---> 04201534b16d
Step 14/17 : RUN ./build --configuration=${config}
 ---> Running in 2575e99d12fd
Index was outside the bounds of the array.
The command '/bin/sh -c ./build --configuration=${config}' returned a non-zero code: 1
```
I have omitted the beginning of the log for the sake of brevity.

**Comment by HebaruSan** [comment]
Comparing https://packages.debian.org/sid/mono-complete to my locally installed copy from mono-project.com, the dependency on `referenceassemblies-pcl` seems to be missing in Debian (making its Mono not all that "complete," ha ha). I think that contains a lot of the runtime libraries, so it's plausible that lacking it could cause this.

Can some Debian users please try to find out what happened to that package? Is it still in the repos somewhere? Is GNU expecting you to download and install it manually? Etc.

Also try installing from https://www.mono-project.com/download/stable/#download-lin-debian . It may appear that "packages are already up-to-date in debian archive," but apparently not without modifications.

**Comment by dkm** [comment]
I've tried installing the `referenceassemblies-pcl`: it was not installed, but CKAN still crashes with the same error messages. I'll try to reinstall from mono project.

**Comment by dkm** [comment]
Ok, here's the results. Installing from mono-project makes ckan work correctly.

After pinning mono-project's packages and installing `mono-complete`:

```
The following NEW packages will be installed:
  libglade2.0-cil libglib2.0-cil libgtk2.0-cil libmono-2.0-1 libmono-2.0-dev libmono-cairo4.0-cil libmono-cecil-private-cil libmono-cil-dev libmono-codecontracts4.0-cil
  libmono-compilerservices-symbolwriter4.0-cil libmono-cscompmgd0.0-cil libmono-csharp4.0c-cil libmono-custommarshalers4.0-cil libmono-data-tds4.0-cil libmono-db2-1.0-cil libmono-debugger-soft4.0a-cil
  libmono-http4.0-cil libmono-management4.0-cil libmono-messaging-rabbitmq4.0-cil libmono-messaging4.0-cil libmono-microsoft-build4.0-cil libmono-microsoft-visualc10.0-cil
  libmono-microsoft-web-infrastructure1.0-cil libmono-oracle4.0-cil libmono-parallel4.0-cil libmono-peapi4.0a-cil libmono-profiler libmono-rabbitmq4.0-cil libmono-relaxng4.0-cil libmono-sharpzip4.84-cil
  libmono-simd4.0-cil libmono-smdiagnostics0.0-cil libmono-system-componentmodel-composition4.0-cil libmono-system-configuration-install4.0-cil libmono-system-data-datasetextensions4.0-cil
  libmono-system-data-entity4.0-cil libmono-system-data-linq4.0-cil libmono-system-data-services-client4.0-cil libmono-system-data-services4.0-cil libmono-system-deployment4.0-cil
  libmono-system-drawing-design4.0-cil libmono-system-dynamic4.0-cil libmono-system-identitymodel-selectors4.0-cil libmono-system-identitymodel4.0-cil libmono-system-io-compression-filesystem4.0-cil
  libmono-system-io-compression4.0-cil libmono-system-json-microsoft4.0-cil libmono-system-json4.0-cil libmono-system-ldap-protocols4.0-cil libmono-system-management4.0-cil libmono-system-messaging4.0-cil
  libmono-system-net-http-formatting4.0-cil libmono-system-net4.0-cil libmono-system-numerics-vectors4.0-cil libmono-system-reactive-core2.2-cil libmono-system-reactive-debugger2.2-cil
  libmono-system-reactive-experimental2.2-cil libmono-system-reactive-interfaces2.2-cil libmono-system-reactive-linq2.2-cil libmono-system-reactive-observable-aliases0.0-cil
  libmono-system-reactive-platformservices2.2-cil libmono-system-reactive-providers2.2-cil libmono-system-reactive-runtime-remoting2.2-cil libmono-system-reactive-windows-forms2.2-cil
  libmono-system-reactive-windows-threading2.2-cil libmono-system-reflection-context4.0-cil libmono-system-runtime-caching4.0-cil libmono-system-runtime-durableinstancing4.0-cil libmono-system-runtime4.0-cil
  libmono-system-servicemodel-activation4.0-cil libmono-system-servicemodel-discovery4.0-cil libmono-system-servicemodel-routing4.0-cil libmono-system-servicemodel-web4.0-cil libmono-system-servicemodel4.0a-cil
  libmono-system-serviceprocess4.0-cil libmono-system-threading-tasks-dataflow4.0-cil libmono-system-web-abstractions4.0-cil libmono-system-web-dynamicdata4.0-cil libmono-system-web-extensions-design4.0-cil
  libmono-system-web-extensions4.0-cil libmono-system-web-http-selfhost4.0-cil libmono-system-web-http-webhost4.0-cil libmono-system-web-http4.0-cil libmono-system-web-mobile4.0-cil
  libmono-system-web-mvc3.0-cil libmono-system-web-razor2.0-cil libmono-system-web-regularexpressions4.0-cil libmono-system-web-routing4.0-cil libmono-system-web-webpages-deployment2.0-cil
  libmono-system-web-webpages-razor2.0-cil libmono-system-web-webpages2.0-cil libmono-system-windows-forms-datavisualization4.0a-cil libmono-system-windows4.0-cil libmono-system-workflow-activities4.0-cil
  libmono-system-workflow-componentmodel4.0-cil libmono-system-workflow-runtime4.0-cil libmono-system-xaml4.0-cil libmono-system-xml-serialization4.0-cil libmono-tasklets4.0-cil libmono-webmatrix-data4.0-cil
  libmono-windowsbase4.0-cil libmonoboehm-2.0-1 libmonosgen-2.0-1 libmonosgen-2.0-dev libnunit-cil-dev libnunit-console-runner2.6.3-cil libnunit-core-interfaces2.6.3-cil libnunit-core2.6.3-cil
  libnunit-framework2.6.3-cil libnunit-mocks2.6.3-cil libnunit-util2.6.3-cil mono-4.0-service mono-complete mono-csharp-shell mono-devel mono-jay mono-llvm-support mono-llvm-tools mono-mcs mono-roslyn
  mono-utils monodoc-base monodoc-browser monodoc-manual msbuild msbuild-libhostfxr msbuild-sdkresolver
The following packages will be DOWNGRADED:
  ca-certificates-mono libmono-accessibility4.0-cil libmono-corlib4.5-cil libmono-i18n-cjk4.0-cil libmono-i18n-mideast4.0-cil libmono-i18n-other4.0-cil libmono-i18n-rare4.0-cil libmono-i18n-west4.0-cil
  libmono-i18n4.0-all libmono-ldap4.0-cil libmono-microsoft-build-engine4.0-cil libmono-microsoft-build-framework4.0-cil libmono-microsoft-build-tasks-v4.0-4.0-cil libmono-microsoft-build-utilities-v4.0-4.0-cil
  libmono-microsoft-csharp4.0-cil libmono-posix4.0-cil libmono-security4.0-cil libmono-sqlite4.0-cil libmono-system-componentmodel-dataannotations4.0-cil libmono-system-configuration4.0-cil
  libmono-system-core4.0-cil libmono-system-data4.0-cil libmono-system-design4.0-cil libmono-system-drawing4.0-cil libmono-system-enterpriseservices4.0-cil libmono-system-ldap4.0-cil
  libmono-system-net-http-webrequest4.0-cil libmono-system-net-http4.0-cil libmono-system-numerics4.0-cil libmono-system-runtime-serialization-formatters-soap4.0-cil libmono-system-runtime-serialization4.0-cil
  libmono-system-security4.0-cil libmono-system-servicemodel-internals0.0-cil libmono-system-transactions4.0-cil libmono-system-web-applicationservices4.0-cil libmono-system-web-services4.0-cil
  libmono-system-web4.0-cil libmono-system-windows-forms4.0-cil libmono-system-xml-linq4.0-cil libmono-system-xml4.0-cil libmono-system4.0-cil libmono-webbrowser4.0-cil libmono-xbuild-tasks4.0-cil mono-4.0-gac
  mono-gac mono-runtime mono-runtime-common mono-runtime-sgen mono-xbuild
```

Not sure what is happening, I have no idea how mono works. What should I report to debian for them to find the issue ?

**Comment by HebaruSan** [comment]
I haven't reported a bug to Debian before that I can recall, so I don't have advice to give. Maybe start with "Mono packaged by Debian doesn't work" and go from there? :wink:

**Comment by dkm** [comment]
https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=920531

**Comment by HebaruSan** [comment]
Thanks @dkm, it looks like they've already got a fix in the works!

```
   * [fcb8084] Add missing libmono-system-native (Closes: #920531)
```

I wonder if `libmono-system-native` is also the cause of #2630...

**Comment by HebaruSan** [comment]
Excellent news! Thanks again for filing that report!
